src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -360,28 +369,57 @@ public void visitQuerySpec(QuerySpec querySpec) {
         if (!querySpec.getGroupByClauseExpressions().isEmpty()) {
             throw new FeatureNotSupportedException("GroupBy not supported");
         }
-        if (querySpec.hasSortSpecifications()) {
-            throw new FeatureNotSupportedException("Sorting not supported");
-        }
         if (querySpec.hasOffsetOrFetchClause()) {
             throw new FeatureNotSupportedException("TODO-HIBERNATE-70 https://jira.mongodb.org/browse/HIBERNATE-70");
         }
 
         var collection = acceptAndYield(querySpec.getFromClause(), COLLECTION_NAME);
 
+        var stages = new ArrayList<AstStage>(3);
+
+        // $match stage
         var whereClauseRestrictions = querySpec.getWhereClauseRestrictions();
-        var filter = whereClauseRestrictions == null || whereClauseRestrictions.isEmpty()
-                ? null
-                : acceptAndYield(whereClauseRestrictions, FILTER);
+        if (whereClauseRestrictions != null && !whereClauseRestrictions.isEmpty()) {
+            var filter = acceptAndYield(whereClauseRestrictions, FILTER);
+            stages.add(new AstMatchStage(filter));
+        }
+
+        // $sort stage
+        getOptionalAstSortStage(querySpec).ifPresent(stages::add);
 
+        // $project stage
         var projectStageSpecifications = acceptAndYield(querySpec.getSelectClause(), PROJECT_STAGE_SPECIFICATIONS);
+        stages.add(new AstProjectStage(projectStageSpecifications));
 
-        var stages = filter == null
-                ? List.of(new AstProjectStage(projectStageSpecifications))
-                : List.of(new AstMatchStage(filter), new AstProjectStage(projectStageSpecifications));
         astVisitorValueHolder.yield(COLLECTION_AGGREGATE, new AstAggregateCommand(collection, stages));
     }
 
+    private Optional<AstSortStage> getOptionalAstSortStage(QuerySpec querySpec) {
+        if (querySpec.hasSortSpecifications()) {
+            var sortFields = new ArrayList<AstSortField>(
+                    querySpec.getSortSpecifications().size());
+            for (SortSpecification sortSpecification : querySpec.getSortSpecifications()) {
+                if (!isFieldPathExpression(sortSpecification.getSortExpression())) {
+                    throw new FeatureNotSupportedException(
+                            format("%s does not support sort key not of field path type", MONGO_DBMS_NAME));
+                }
+                if (sortSpecification.getNullPrecedence() != null
+                        && sortSpecification.getNullPrecedence() != NullPrecedence.NONE) {
+                    throw new FeatureNotSupportedException(format(
+                            "%s does not support nulls precedence: NULLS %s",
+                            MONGO_DBMS_NAME, sortSpecification.getNullPrecedence()));
```

Comment:
MongoDB puts `null` to the top in ASC case (only after `MinKey `) as per https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#std-label-bson-types-comparison-order

However, there is no counterpart of HQL's `NULLS LAST` and I am not sure whether the `NULLS FIRST` corresponds to Mongo's behaviour.  My understanding is we could simply throw exception to avoid confusion.

Another alternative is to only throw exception when `NULLS LAST` is specified in HQL, provided `NULLS FIRST` is just what Mongo is doing. Not sure whether the precondition holds or not.